### PR TITLE
Add reducers testing of articles,  header, and index

### PR DESCRIPTION
**Pull request by YuCJ**
- Add header reducer testing

### CODE REVIEW

**Comment by nickhsine** [comment]
💯 👍

**Comment by YuCJ** [comment]
- Add index reducers testing:
Includes entities, articleSlugToId, and topicSlugToId reducers

**Comment by YuCJ** [comment]
- Add `articles` reducers testing:
  Includes `relatedArticles`, `articlesByUuids`, and `featureArticles` reducers.

File: src/reducers/__test__/articles.test.js [hunk 1]
```diff
@@ -0,0 +1,459 @@
...
+  [types.FETCH_RELATED_ARTICLES_REQUEST]: {
+    type: types.FETCH_RELATED_ARTICLES_REQUEST,
+    id: 'act_article_id_01',
+    relatedIds: [ 'act_related_id_01' ],
+    url: 'act_api_url_01'
+  },
+  [types.FETCH_RELATED_ARTICLES_FAILURE]: {
+    type: types.FETCH_RELATED_ARTICLES_FAILURE,
+    id: 'act_article_id_01',
+    relatedIds: [ 'act_related_article_id_01' ],
```
Comment (nickhsine) [inline src/reducers/__test__/articles.test.js]: remove `relatedIds` here?

File: src/reducers/__test__/articles.test.js [hunk 2]
```diff
@@ -0,0 +1,459 @@
...
+      })
+    })
+    it('SHOULD return expected state(on previous state) WHEN previous state exist', () => {
+      expect(
+        reducers.relatedArticles(mockPreviousState_relatedArticles, action)
+      ).to.deep.equal(mockExpectedStateOnPre_relatedArticles.success)
+    })
+  })
+})
+
```
Comment (nickhsine) [inline src/reducers/__test__/articles.test.js]: You can separate the `mockActions` and `mockExpectedState` into different `describe`.

File: src/reducers/__test__/header.test.js [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+  },
+  [types.SET_ARTICLE_TOPIC_LIST]: {
+    type: types.SET_ARTICLE_TOPIC_LIST,
+    topicArr: [ 'Mock Action: Topic C', 'Mock Action: Topic D' ]
+  },
+  [types.SET_BOOKMARKS_OF_LONGFORM_ARTICLE]: {
+    type: types.SET_BOOKMARKS_OF_LONGFORM_ARTICLE,
+    bookmarks: [ 'Mock Action: Bookmark C', 'Mock Action: Bookmark D' ]
+  }
+}
```
Comment (nickhsine) [inline src/reducers/__test__/header.test.js]: Split these mock data into different `describe`

File: src/reducers/__test__/articles.test.js (hunk 1 above)
Comment (YuCJ) [inline src/reducers/__test__/articles.test.js]: The reducer will not save `relatedIds` into state, but it's included in input action

`actions/articles.js`:

```
function failToReceiveRelatedArticles(id, relatedIds, error) {
  return {
    type: types.FETCH_RELATED_ARTICLES_FAILURE,
    id,
    relatedIds,
    error,
    failedAt: Date.now()
  }
}
```

**Comment by YuCJ** [comment]
[Update] Split mock data into different describe

**Comment by nickhsine** [comment]
👍 :shipit:

**Pull request merged**